src/storage/src/manifest/storage.rs

```diff
@@ -352,28 +395,61 @@ impl ManifestLogStorage for ManifestObjectStore {
         &self,
         version: ManifestVersion,
     ) -> Result<Option<(ManifestVersion, Vec<u8>)>> {
-        let path = self.checkpoint_file_path(version);
-        match self.object_store.read(&path).await {
-            Ok(checkpoint) => Ok(Some((version, checkpoint))),
-            Err(e) if e.kind() == ErrorKind::NotFound => Ok(None),
-            Err(e) => Err(e).context(ReadObjectSnafu { path }),
+        let entry = self
+            .get_paths(|entry| {
+                let file_name = entry.name();
+                if is_checkpoint_file(file_name) && version == file_version(file_name) {
+                    return Some(entry.path().to_string());
+                }
+                None
+            })
+            .await?;
```

Comment:
This is costly. I think we could store the compression type in the `CheckpointMetadata`. Then we could get the correct checkpoint path.
https://github.com/GreptimeTeam/greptimedb/blob/224ec9bd259277c915b11d4a6dafab20c5ee7593/src/storage/src/manifest/storage.rs#L129-L135

We might need to add an argument `compress_type` to `load_checkpoint()` and `delete_checkpoint()`.

Reply:
Is there possible user store multiple different compress file? Such as due to some reason the compress algorithm is changed. v1,v2,v3 was compress by gzip, v4 was not compress. It may lead a error. How about add an argument `compress_type` to `load_checkpoint()` and `delete_checkpoint()`. In most case it will find file correctly, but if we don't find file according to suggest compress_type, we fall back to scan the folder

Reply:
> Such as due to some reason the compress algorithm is changed. v1,v2,v3 was compress by gzip, v4 was not compress. It may lead a error.

If we store the compress type into `CheckpointMetadata`, this issue maybe gone. We can always use the correct compress type to decompress the snapshot.

> In most case it will find file correctly, but if we don't find file according to suggest compress_type, we fall back to scan the folder

Yep, the fallback is necessary.

Reply:
The fallback should be necessary if we want to support different compression methods for manifest. Another approach is once a compression type is chosen, we always use that compression type (stored in CheckpointMetadata) to compress the manifest.

But is it beneficial to provide so many compression types for this? I think over 99% of users won't change the default compression type for manifest. We could always use gzip to compress the manifest, thus we don't need to consider this problem. @killme2008 @Taylor-lagrange

Reply:
There is another problem with compression type stored in `CheckpointMetadata`，If the manifest version was too small for no checkpoint generated, we will eventually fall back to scan the whole folder (Unless we record compression type change in each action, but I don't recommend this as it complicates a very simple problem).

I agree with @evenyag that it's should be a simple question. How about only support GZ compress (if we really want to change the  compression type,  just replace some const definition and recompile the database).

 All delete related function we can delete such 000.json.gz, 000.json.zstd(even if it not exsit).  In the end, no matter what compression algorithm this file uses, it will be deleted.  which make it can support all compression type.

As for load operation, we load path by Default compression type and followed by other compress type one by one(if Default compression type failed), for most case the Default compression type is enough. In the meantime， we have enough backward compatibility.

Reply:
The Default compression type is a const variable in rust code, so we don't need to record it in `CheckpointMetadata`

Reply:
And there is no need to expose too many low-level implementation details to the upper layer

Reply:
Gzip is enough. We use the `.json.gz` file by default and fallback to `.json` if it doesn't exist.